subsys/zbus/zbus.c

```diff
@@ -64,12 +117,28 @@ static inline int _zbus_vded_exec(const struct zbus_channel *chan, k_timepoint_t
 	int err = 0;
 	int last_error = 0;
 
-	_ZBUS_ASSERT(chan != NULL, "chan is required");
-
 	/* Static observer event dispatcher logic */
 	struct zbus_channel_observation *observation;
 	struct zbus_channel_observation_mask *observation_mask;
 
+#if defined(CONFIG_ZBUS_MSG_SUBSCRIBER)
+	struct net_buf *buf =
+		_zbus_create_net_buf(&_zbus_msg_subscribers_pool,
+				     sizeof(const struct zbus_channel *) + zbus_chan_msg_size(chan),
+				     sys_timepoint_timeout(end_time));
+
+	_ZBUS_ASSERT(buf != NULL, "net_buf zbus_msg_subscribers_pool is "
+				  "unavailable or heap is full");
+
+	net_buf_add_mem(buf, zbus_chan_msg(chan), zbus_chan_msg_size(chan));
+	net_buf_add_mem(buf, &chan, sizeof(const struct zbus_channel *));
+#else
+	struct net_buf *buf = NULL;
```

Comment:
declare `struct net_buf` separately, bit odd to have this initialized in a condition and referenced later.

Reply:
It is weird. Changed.